datamations/datamation_frame.py

```diff
@@ -289,27 +399,36 @@ def datamation_sanddance(self):
                             onchange="window.%s.onSlide('%s')"
                         />
                         </div>
-                    </div>
-                    <div class="description"></div>
+                        <div class="button-wrap">
+                            <button class="export-btn" onclick="%s.exportGif(true)">
+                                <i class="fas fa-download"></i>
+                            </button>
+                        </div>
                     </div>
                     
-                    <div class="vega-vis-wrapper">
-                    <div class="vega-for-axis"></div>
-                    <div class="vega-other-layers"></div>
-                    <div class="vega-vis"></div>
+                    <div class="export-wrapper">
+                        <div class="description"></div>
+                        <div class="vega-vis-wrapper">
+                            <div class="vega-for-axis"></div>
+                            <div class="vega-other-layers"></div>
+                            <div class="vega-vis"></div>
+                        </div>
                     </div>
                 </div>
                 </div>
                 `));
 
-                require(['d3', 'vega', 'vega-util', 'vega-lite', 'vega-embed', 'gemini', 'app', 'utils', 'layout', 'config', 'hack-facet-view'], function(d3, vega, vegaUtil, vegaLite, vegaEmbed, gemini, app, utils, layout, config, hackFacetView) {
+                require(['d3', 'vega', 'vega-util', 'vega-lite', 'vega-embed', 'gemini', 'html2canvas', 'gifshot', 'download2', 'app', 'utils', 'layout', 'config', 'hack-facet-view', 'custom-animations'], function(d3, vega, vegaUtil, vegaLite, vegaEmbed, gemini, html2canvas, gifshot, download2, app, utils, layout, config, hackFacetView, customAnimations) {
                     window.d3 = d3
                     window.vegaEmbed = vegaEmbed
                     window.gemini = gemini
+                    window.html2canvas = html2canvas
+                    window.gifshot = gifshot
+                    window.CustomAnimations = customAnimations
                     window.%s = App("%s", {specs: %s, autoPlay: true});
```

Comment:
I think we will need to merge in the changes to this that call `App` as a method of the datamations module, e.g. `datamations.App` and update the `require` call accordingly.